### PR TITLE
New Entry: Java Generics

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

From Target Keyword List.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/generics/generics.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  - 'Data Types'
+  - 'Constructors'
+  - 'Methods'
+  - 'OOP'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Generics** refer to the ability to use a type as a parameter to methods and classes. This provides the ability to define a set of related classes or methods that can operate on many different types with a single declaration. This also allows type safety at compile-time allowing invalid types to be caught during compilation.
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:18]:
```suggestion
**Generics** refer to the ability to use a [type](https://www.codecademy.com/resources/docs/java/data-types) as a parameter to [methods](https://www.codecademy.com/resources/docs/java/methods) and [classes](https://www.codecademy.com/resources/docs/java/classes). This provides the ability to define a set of related classes or methods that can operate on many different types with a single declaration. This also allows type safety at compile-time allowing invalid types to be caught during compilation.
```

File: content/java/concepts/generics/generics.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+**Note:** Java type parameters can only be reference types, not primitive types (like `int`, `double` or `char`).
+
+## Generic Methods
+
+A generic method declaration can be called with arguments of different types. The compiler handles each method call appropriately based on the types of the arguments.
+
+### Syntax
+
+```pseudo
+
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:29]:
```suggestion
```pseudo
```

File: content/java/concepts/generics/generics.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+<T> void myMethod( T argument ) {
+  // Method body
+  }
+```
+
+The type parameter section is before the return type and is enclosed in angle brackets `<...>`. Inside are one or more type parameters separated by commas (`T` above) each one is a identifier for a specific type name. The identifiers can used to declare the return type, the types of the arguments passed to the method, and as type specifiers within the method body.
+
+### Example
+
+The following example shows how a generic method can accept arrays of different types.
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:39]:
```suggestion
The following example shows how a generic method can accept [arrays](https://www.codecademy.com/resources/docs/java/arrays) of different types:
```

File: content/java/concepts/generics/generics.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+A generic method declaration can be called with arguments of different types. The compiler handles each method call appropriately based on the types of the arguments.
+
+### Syntax
+
+```pseudo
+
+<T> void myMethod( T argument ) {
+  // Method body
+  }
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:32]:
```suggestion
<T> void myMethod( T argument ) {
  // Method body
}
```

File: content/java/concepts/generics/generics.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+```java
+public class GenericMethodExample {
+  // Generic method
+  public static <T> void arrayPrint( T[] a ) {
+    for(T item : a) {
+      System.out.printf("%s",item);
+      System.out.println();
+    }
+  }
+  // Use example
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:50]:
```suggestion
  }
  
  // Use example
```

File: content/java/concepts/generics/generics.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+N
+E
+R
+I
+C
+```
+
+## Generic Classes
+
+A generic class takes one or more type parameters when it is declared. One of the more familiar uses of a generic class is with Java's various collection classes. An `ArrayList` for instance is a generic class. When it is declared it requires a type parameter to specify what kind of type the `ArrayList` contains. For example:
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:83]:
```suggestion
A generic class takes one or more type parameters when it is declared. One of the more familiar uses of a generic class is with Java's various collection classes. An [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list) for instance is a generic class. When it is declared it requires a type parameter to specify what kind of type the `ArrayList` contains. For example:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Left a few comments/suggestions for you 😄

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented your suggestions.

File: content/java/concepts/generics/generics.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  - 'Methods'
+  - 'OOP'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Generics** refer to the ability to use a [type](https://www.codecademy.com/resources/docs/java/data-types) as a parameter to [methods](https://www.codecademy.com/resources/docs/java/methods) and [classes](https://www.codecademy.com/resources/docs/java/classes). This provides the ability to define a set of related classes or methods that can operate on many different types with a single declaration. This also allows type safety at compile-time allowing invalid types to be caught during compilation.
+
+**Note:** Java type parameters can only be reference types, not primitive types (like `int`, `double` or `char`).
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:20]:
```suggestion
**Note:** Java type parameters can only be reference types, not primitive types (like `int`, `double`, or `char`).
```

File: content/java/concepts/generics/generics.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+Description: 'Generics refer to the ability to use a type as a parameter to methods and classes.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Classes'
+  - 'Objects'
+  - 'Data Types'
+  - 'Constructors'
+  - 'Methods'
+  - 'OOP'
```
Comment (Dusch4593) [inline content/java/concepts/generics/generics.md:12]:
```suggestion
  - 'OOP'
  - 'Parameters'
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Just a few last-minute comments/suggestions

**Comment by SSwiniarski** [comment]
@Dusch4593, I committed those changes.

**Review by KTom101 — COMMENTED** [review]
@SSwiniarski I only see a minor change and have one comment.

File: content/java/concepts/generics/generics.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  - 'Parameters'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Generics** refer to the ability to use a [type](https://www.codecademy.com/resources/docs/java/data-types) as a parameter to [methods](https://www.codecademy.com/resources/docs/java/methods) and [classes](https://www.codecademy.com/resources/docs/java/classes). This provides the ability to define a set of related classes or methods that can operate on many different types with a single declaration. This also allows type safety at compile-time allowing invalid types to be caught during compilation.
+
+**Note:** Java type parameters can only be reference types, not primitive types (like `int`, `double`, or `char`).
```
[the remark this answers is not in the document]
Comment (SSwiniarski) [reply content/java/concepts/generics/generics.md:21]: Might be worth opening an issue. @Dusch4593, thoughts?

**Comment by SSwiniarski** [comment]
@KTom101, I applied your suggestion.

**Review by KTom101 — APPROVED** [review]

File: content/java/concepts/generics/generics.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/java/concepts/generics/generics.md:21]: Excellent suggestion! I just [filed an issue](https://github.com/Codecademy/docs/issues/850) 👍🏻

**Pull request merged**